### DISCUSSION TITLE
Limitations of Mathematical Transformations

### DISCUSSION THREAD

**Discussion by AhmedThahir** [OTexts/Discussion#5] [General]
Book Reference: https://otexts.com/fpp3/transformations.html#mathematical-transformations

Dear moderators
Performing a linear regression with transformation, such as log transform, will be optimizing for minimizing the residual:
$\vert \log \hat y - \log y \vert$ instead of the intended $\vert \hat y - y \vert$ 

From my understanding, this will result in under-forecasting as the loss function as $\vert \log \hat y - \log y \vert = \log \left\vert \dfrac{\hat y}{y} \right\vert$, which will reduce the importance of large response values and increase the importance of small response values. 

This is also cited in https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2021JG006238:
> Log-transforming the dependent variable of a regression model, though convenient and frequently used, is accompanied by an under-prediction problem. We found that this underprediction can reach up to 20%, which is significant in studies that aim to estimate annual budgets

As esteemed experts in this field, I would like to know
1. Your thoughts on this problem
2. Does this really have impact in practice 
3. Your suggestions on alleviating this

Best regards

**Comment by robjhyndman** [comment]
This is addressed in Section 5.6: https://otexts.com/fpp3/ftransformations.html#bias-adjustments

Yes, it has an impact in practice. The fable package will automatically do the bias adjustment for any transformation used.

**Reply by AhmedThahir** [reply]
Noted, Prof Hyndman. I have gone through it, thank you! 

However, the matter of weighting is not mentioned here - ie due to the log transformation, larger response values are given lower weight than smaller response values. I am not sure if this affects the closed-form solution, but it definitely would affect the solution obtained through iterative optimization. 

From an optimization perspective, optimizing MSE for a log transformation would be equivalent to optimizing RMSLE for non-transformed.

This is RMSLE:
![images](https://github.com/user-attachments/assets/c14fa2d5-477e-4198-96de-dcf9d627fa6b)

> RMSLE incurs a larger penalty for the underestimation of the Actual variable than the Overestimation. In simple words, more penalty is incurred when the predicted Value is less than the Actual Value. On the other hand, Less penalty is incurred when the predicted value is more than the actual value.
> https://medium.com/analytics-vidhya/root-mean-square-log-error-rmse-vs-rmlse-935c6cc1802a

Since I mainly work with custom loss functions, I utilise iterative optimization and I would like to understand how to alleviate this. 

I would say maybe add a sample weight as $y_i$ to counteract?

**Reply by robjhyndman** [reply]
Changing the loss function would tackle a different problem. We do not consider that in our book.

The approach we take is to do a bias adjustment to avoid the under- or over-prediction that arises due to a transformation being applied to the data.

**Reply by AhmedThahir** [reply]
> Changing the loss function would tackle a different problem. We do not consider that in our book.

Would it be possible to share any resources for me to read more on that? 

> The approach we take is to do a bias adjustment to avoid the under- or over-prediction that arises due to a transformation being applied to the data.

This adjustment completely eliminates the need for updating the loss function. Understood.

---

Also, would this bias adjustment be required for any type of GLM (Generalized Linear Model)?
